<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Single-cell Mitochondrial Mutation Analysis Tool â€¢ scMitoMut</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="bootstrap-toc.css">
<script src="bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="pkgdown.css" rel="stylesheet">
<script src="pkgdown.js"></script><meta property="og:title" content="Single-cell Mitochondrial Mutation Analysis Tool">
<meta property="og:description" content="
    This package is designed for analyzing mitochondrial mutations using
    single-cell sequencing data, such as scRNASeq and scATACSeq (preferably the
    latter due to RNA editing issues). It includes functions for mutation
    filtering and visualization. In the future, the visualization tool will
    become an independent package. Mutation filtering is performed by fitting a
    statistical model to account for various sources of noise, including PCR
    error, sequencing error, mtDNA sampling and/or heteroplasmy dynamics. The
    model tests whether the observed allele frequency of a locus in a cell can
    be explained by the noise model. If not, we classify it as a mutation. The
    input for this analysis is the allele frequency. The noise model consists
    of three independent models: binomial, binomial-mixture, and beta-binomial
    models.">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-home">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="index.html">scMitoMut</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.99.2</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="articles/Analysis_colon_cancer_dataset.html">scMitoMut demo: CRC dataset</a>
    </li>
  </ul>
</li>
<li>
  <a href="news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="contents col-md-9">
<div class="section level1">
<div class="page-header"><h1 id="scmitomut-an-r-package-for-mitochondrial-mutation-analysis-in-single-cell-omics-data">scMitoMut: An R Package for Mitochondrial Mutation Analysis in Single-Cell Omics Data<a class="anchor" aria-label="anchor" href="#scmitomut-an-r-package-for-mitochondrial-mutation-analysis-in-single-cell-omics-data"></a>
</h1></div>
<p><a href="https://github.com/wenjie1991/scMitoMut/actions/workflows/r.yml" class="external-link"><img src="https://github.com/wenjie1991/scMitoMut/actions/workflows/r.yml/badge.svg" alt="R"></a></p>
<div class="section level2">
<h2 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h2>
<p>We are currently submitting the package to Bioconductor. To try it out, you can use the source code provided.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># install.packages("devtools")</span></span>
<span><span class="fu">devtools</span><span class="fu">::</span><span class="fu"><a href="https://remotes.r-lib.org/reference/install_github.html" class="external-link">install_github</a></span><span class="op">(</span><span class="st">"username/packagename"</span>, build_vignettes <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="vignette">Vignette<a class="anchor" aria-label="anchor" href="#vignette"></a>
</h2>
<p>The vignette can be found here: <a href="">https://github.com/TeamPerie/scMitoMut/blob/main/vignettes/Analysis_colon_cancer_dataset.Rmd</a>.</p>
<p>You can also access the HTML version by using the R command <code>browseVignettes('scMitoMut')</code> after installing the package.</p>
</div>
<div class="section level2">
<h2 id="mini-example">Mini Example<a class="anchor" aria-label="anchor" href="#mini-example"></a>
</h2>
<p>This is a simple example that demonstrates the main function of the package. It can be executed in less than 1 minute.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">scMitoMut</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># load the data</span></span>
<span><span class="co">## Use the example data</span></span>
<span><span class="va">f</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"mini_dataset.tsv.gz"</span>, package <span class="op">=</span> <span class="st">"scMitoMut"</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Load the data with parse_table function</span></span>
<span><span class="va">f_h5</span> <span class="op">=</span> <span class="fu"><a href="reference/parse_table.html">parse_table</a></span><span class="op">(</span><span class="va">f</span>, sep <span class="op">=</span> <span class="st">"\t"</span>, h5_file <span class="op">=</span> <span class="st">"./mut.h5"</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## open the h5f file</span></span>
<span><span class="va">x</span> <span class="op">=</span> <span class="fu"><a href="reference/open_h5_file.html">open_h5_file</a></span><span class="op">(</span><span class="va">f_h5</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># run the model fit</span></span>
<span><span class="co"># You can increase the cpu core to accelerate</span></span>
<span><span class="co"># This step need some time, so the result will be kept in h5 file,</span></span>
<span><span class="co"># you do not need to re-run this step, when you load the h5 file next time.</span></span>
<span><span class="fu"><a href="reference/run_model_fit.html">run_model_fit</a></span><span class="op">(</span><span class="va">x</span>, mc.cores <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Filter the loci based on the model fit results</span></span>
<span><span class="co"># The filter options will be keeped in the object by memory</span></span>
<span><span class="co"># Next time you re-load the h5 file, the filter will be initiated as default</span></span>
<span><span class="va">x</span> <span class="op">=</span> <span class="fu"><a href="reference/filter_loc.html">filter_loc</a></span><span class="op">(</span><span class="va">x</span>, </span>
<span>    min_cell <span class="op">=</span> <span class="fl">10</span>, </span>
<span>    model <span class="op">=</span> <span class="st">"bb"</span>, </span>
<span>    p_threshold <span class="op">=</span> <span class="fl">0.01</span>, </span>
<span>    p_adj_method <span class="op">=</span> <span class="st">"fdr"</span></span>
<span>    <span class="op">)</span></span>
<span><span class="va">x</span></span>
<span></span>
<span><span class="co"># Set the cell annotation</span></span>
<span><span class="va">f</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"mini_dataset_cell_ann.csv"</span>, package <span class="op">=</span> <span class="st">"scMitoMut"</span><span class="op">)</span></span>
<span><span class="va">cell_ann</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.csv</a></span><span class="op">(</span><span class="va">f</span>, row.names<span class="op">=</span><span class="fl">1</span><span class="op">)</span></span>
<span><span class="co"># Prepare the color for cell annotation</span></span>
<span><span class="va">colors</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span></span>
<span>    <span class="st">"Cancer Epi"</span> <span class="op">=</span> <span class="st">"#f28482"</span>,</span>
<span>    Blood <span class="op">=</span> <span class="st">"#f6bd60"</span><span class="op">)</span></span>
<span><span class="va">ann_colors</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="st">"SeuratCellTypes"</span> <span class="op">=</span> <span class="va">colors</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># plot the heatmap for binary mutation</span></span>
<span><span class="fu"><a href="reference/plot_heatmap.html">plot_heatmap</a></span><span class="op">(</span><span class="va">x</span>, type <span class="op">=</span> <span class="st">"binary"</span>, cell_ann <span class="op">=</span> <span class="va">cell_ann</span>, ann_colors <span class="op">=</span> <span class="va">ann_colors</span>, percent_interp <span class="op">=</span> <span class="fl">0.2</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># plot the heatmap for p-value</span></span>
<span><span class="fu"><a href="reference/plot_heatmap.html">plot_heatmap</a></span><span class="op">(</span><span class="va">x</span>, type <span class="op">=</span> <span class="st">"p"</span>, cell_ann <span class="op">=</span> <span class="va">cell_ann</span>, ann_colors <span class="op">=</span> <span class="va">ann_colors</span>, percent_interp <span class="op">=</span> <span class="fl">0.2</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># plot the heatmap for allele frequency</span></span>
<span><span class="fu"><a href="reference/plot_heatmap.html">plot_heatmap</a></span><span class="op">(</span><span class="va">x</span>, type <span class="op">=</span> <span class="st">"af"</span>, cell_ann <span class="op">=</span> <span class="va">cell_ann</span>, ann_colors <span class="op">=</span> <span class="va">ann_colors</span>, percent_interp <span class="op">=</span> <span class="fl">0.2</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># check af~coverage for one loci</span></span>
<span><span class="fu"><a href="reference/plot_af_coverage.html">plot_af_coverage</a></span><span class="op">(</span><span class="va">x</span>, <span class="st">"chrM.200"</span><span class="op">)</span></span></code></pre></div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <div class="license">
<h2 data-toc-skip>License</h2>
<ul class="list-unstyled">
<li><a href="LICENSE.html">Full license</a></li>
<li><small><a href="https://opensource.org/licenses/mit-license.php" class="external-link">MIT</a> + file <a href="LICENSE-text.html">LICENSE</a></small></li>
</ul>
</div>


<div class="citation">
<h2 data-toc-skip>Citation</h2>
<ul class="list-unstyled">
<li><a href="authors.html#citation">Citing scMitoMut</a></li>
</ul>
</div>

<div class="developers">
<h2 data-toc-skip>Developers</h2>
<ul class="list-unstyled">
<li>Wenjie Sun <br><small class="roles"> Maintainer, author </small> <a href="https://orcid.org/0000-0002-3100-2346" target="orcid.widget" aria-label="ORCID" class="external-link"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a> </li>
<li><a href="authors.html">More about authors...</a></li>
</ul>
</div>



  </div>
</div>


      <footer><div class="copyright">
  <p></p>
<p>Developed by Wenjie Sun.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
