<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>scMitoMut demo: CRC dataset • scMitoMut</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="scMitoMut demo: CRC dataset">
<meta property="og:description" content="scMitoMut">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">scMitoMut</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.99.2</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/Analysis_colon_cancer_dataset.html">scMitoMut demo: CRC dataset</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>scMitoMut demo: CRC dataset</h1>
                        <h4 data-toc-skip class="author">Wenjie Sun</h4>
                        
      
      
      <div class="hidden name"><code>Analysis_colon_cancer_dataset.Rmd</code></div>

    </div>

    
    
<p>Install package:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://bioconductor.github.io/BiocManager/reference/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">"scMitoMut"</span><span class="op">)</span></span></code></pre></div>
<p>Load packages:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">scMitoMut</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://r-datatable.com" class="external-link">data.table</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org" class="external-link">ggplot2</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/grimbough/rhdf5" class="external-link">rhdf5</a></span><span class="op">)</span></span></code></pre></div>
<div class="section level2">
<h2 id="overview">Overview<a class="anchor" aria-label="anchor" href="#overview"></a>
</h2>
<div class="section level3">
<h3 id="key-functions">Key functions<a class="anchor" aria-label="anchor" href="#key-functions"></a>
</h3>
<p>Key functions used in the scMitoMut package:</p>
<ul>
<li>
<code><a href="../reference/parse_mgatk.html">parse_mgatk()</a></code>: Parses the mgatk output and saves the
result in an H5 file.</li>
<li>
<code><a href="../reference/open_h5_file.html">open_h5_file()</a></code>: Opens the H5 file and returns a
“mtmutObj” object.</li>
<li>
<code><a href="../reference/subset_cell.html">subset_cell()</a></code>: Subsets the cells in the mtmutObj
object.</li>
<li>
<code><a href="../reference/run_model_fit.html">run_model_fit()</a></code>: Runs the model fitting and saves the
results in the H5 file.</li>
<li>
<code>mut_filter()</code>: Filters the mutations based on specific
criteria.</li>
<li>
<code><a href="../reference/plot_heatmap.html">plot_heatmap()</a></code>: Plots a heatmap with different types of
data, including p-values, allele frequencies, or binary mutation
status.</li>
<li>
<code><a href="../reference/export_dt.html">export_df()</a></code>, <code><a href="../reference/export_dt.html">export_af()</a></code>,
<code><a href="../reference/export_dt.html">export_pval()</a></code>, and <code><a href="../reference/export_dt.html">export_binary()</a></code>: Export the
mutation data in various formats.</li>
</ul>
</div>
<div class="section level3">
<h3 id="key-conceptions">Key conceptions<a class="anchor" aria-label="anchor" href="#key-conceptions"></a>
</h3>
<ul>
<li>Somatic mutation: a mutation that occurs in a cell or a cell
population, but not in the germline.</li>
<li>AF or VAF: allele frequency or variant allele frequency. It is the
ratio of the number of reads supporting the variant allele to the total
number of reads.</li>
<li>p-value: the probability of observing a test statistic at least as
extreme as the one that was observed, assuming that the null hypothesis
is correct.</li>
</ul>
<p><strong>IMPORTANT</strong>: In this vignette, I will use the term
“mutation” to refer to the somatic mutation. For the somatic mutation, I
use the dominant allele detected as the reference allele. And if the
reference allele frequency is significant (FDR &lt; 0.05) lowe, I will
call the <strong>locus</strong> a mutation.</p>
</div>
<div class="section level3">
<h3 id="background">Background<a class="anchor" aria-label="anchor" href="#background"></a>
</h3>
<p>Single-cell genomics technology provides a powerful tool for
understanding cellular heterogeneity and diversity within complex
tissues.</p>
<p>Mitochondrial DNA (mtDNA) is small, without chromatin, and has
multiple copies within a cell. Those features help us to get good mtDNA
sequencing depth in single-cell ATAC seq data, which facilitates somatic
mutation detections.</p>
<p>Together with chromatin openness data, mitochondrial somatic mutation
can be used as a lineage label for helping us understand cellular
differentiation, oncogenesis and development.</p>
<p>In this vignette, the scMitoMut package is used to identify and
visualize mtDNA single nucleic somatic mutations.</p>
<p>In the following analysis, scMitoMut was used to analyze the allele
count data, which is the output of <a href="https://github.com/caleblareau/mgatk" class="external-link">mgatk</a>. The only few loci
have been selected for the demonstration purpose to reduce the file size
and the run time.</p>
<p>The full demostration can be found XXX, which shows the analysis
begins with mgatk and CellRanger output.</p>
</div>
</div>
<div class="section level2">
<h2 id="loading-data">Loading data<a class="anchor" aria-label="anchor" href="#loading-data"></a>
</h2>
<p>We load the allele count table with the <code>parse_table</code>
function. The allele count table consists with following columns:</p>
<ol style="list-style-type: decimal">
<li>
<code>loc</code>: the locus name</li>
<li>
<code>cell_barcode</code>: the cell barcode of the cell</li>
<li>
<code>fwd_depth</code>: the forward read count of the allele</li>
<li>
<code>rev_depth</code>: the reverse read count of the allele</li>
<li>
<code>alt</code>: the allele name</li>
<li>
<code>coverage</code>: the total read count of the locus</li>
<li>
<code>ref</code>: the reference allele name</li>
</ol>
<p>Instead of using the table above as input, the output from the
<code>mgatk</code> also can be read in directly using the
<code>parse_mgatk</code> function.</p>
<p>Using the <code>parse_table</code> function or
<code>parse_mgatk</code> function, the allele count data are transformed
into an <code>H5</code> file. The <code>H5</code> file works as a
database, which does not occupy the memory, and data can be accessed by
querying. It helps for better memory usage and faster loading.</p>
<p>The process may take some minutes. The return value is the
<code>H5</code> file path.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Load the allele count table</span></span>
<span><span class="va">f</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"mini_dataset.tsv.gz"</span>, package <span class="op">=</span> <span class="st">"scMitoMut"</span><span class="op">)</span></span>
<span><span class="va">f_h5_tmp</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempfile</a></span><span class="op">(</span>fileext <span class="op">=</span> <span class="st">".h5"</span><span class="op">)</span></span>
<span><span class="va">f_h5</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/parse_table.html">parse_table</a></span><span class="op">(</span><span class="va">f</span>, h5_file <span class="op">=</span> <span class="va">f_h5_tmp</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">f_h5</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] "/var/folders/46/vh_b6qzs5kzgdhvx1kcn7hyh0000gn/T//RtmpqTYBud/file18985cd45930.h5"</span></span></code></pre>
<p>Once we have the <code>H5</code> file, we can use
<code>open_h5_file</code> function to load it and get an object named
“mtmutObj”.</p>
<p><strong>Detail</strong>: On this step, the <code>mtmutObj</code> has
6 slots - <code>h5f</code> is the <code>H5</code> file handle -
<code>mut_table</code> is the allele count table - <code>loc_list</code>
is a list of available loci - <code>loc_selected</code> is the selected
loci - <code>cell_list</code> is a list of available cell ids -
<code>cell_selected</code> is the selected cell ids</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Open the h5 file as a scMitoMut object</span></span>
<span><span class="va">x</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/open_h5_file.html">open_h5_file</a></span><span class="op">(</span><span class="va">f_h5</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/str.html" class="external-link">str</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## List of 9</span></span>
<span><span class="co">##  $ file         : chr "/var/folders/46/vh_b6qzs5kzgdhvx1kcn7hyh0000gn/T//RtmpqTYBud/file18985cd45930.h5"</span></span>
<span><span class="co">##  $ h5f          :Formal class 'H5IdComponent' [package "rhdf5"] with 2 slots</span></span>
<span><span class="co">##   .. ..@ ID    : chr "72057594037927938"</span></span>
<span><span class="co">##   .. ..@ native: logi FALSE</span></span>
<span><span class="co">##  $ mut_table    :Formal class 'H5IdComponent' [package "rhdf5"] with 2 slots</span></span>
<span><span class="co">##   .. ..@ ID    : chr "144115188075855873"</span></span>
<span><span class="co">##   .. ..@ native: logi FALSE</span></span>
<span><span class="co">##  $ loc_list     : chr [1:16(1d)] "chrM.200" "chrM.204" "chrM.310" "chrM.824" ...</span></span>
<span><span class="co">##  $ loc_selected : chr [1:16(1d)] "chrM.200" "chrM.204" "chrM.310" "chrM.824" ...</span></span>
<span><span class="co">##  $ cell_list    : chr [1:1359(1d)] "AAACGAAAGAACCCGA-1" "AAACGAAAGTACCTCA-1" "AAACGAACAGAAAGAG-1" "AAACGAAGTGGTCGAA-1" ...</span></span>
<span><span class="co">##  $ cell_selected: chr [1:1359(1d)] "AAACGAAAGAACCCGA-1" "AAACGAAAGTACCTCA-1" "AAACGAACAGAAAGAG-1" "AAACGAAGTGGTCGAA-1" ...</span></span>
<span><span class="co">##  $ loc_pass     : NULL</span></span>
<span><span class="co">##  $ loc_filter   :List of 4</span></span>
<span><span class="co">##   ..$ min_cell    : num 1</span></span>
<span><span class="co">##   ..$ model       : chr "bb"</span></span>
<span><span class="co">##   ..$ p_threshold : num 0.05</span></span>
<span><span class="co">##   ..$ p_adj_method: chr "fdr"</span></span>
<span><span class="co">##  - attr(*, "class")= chr "mtmutObj"</span></span></code></pre>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Show what's in the h5 file</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/rhdf5/man/h5ls.html" class="external-link">h5ls</a></span><span class="op">(</span><span class="va">x</span><span class="op">$</span><span class="va">h5f</span>, recursive <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##   group          name       otype dclass  dim</span></span>
<span><span class="co">## 0     /     cell_list H5I_DATASET STRING 1359</span></span>
<span><span class="co">## 1     / cell_selected H5I_DATASET STRING 1359</span></span>
<span><span class="co">## 2     /      loc_list H5I_DATASET STRING   16</span></span>
<span><span class="co">## 3     /  loc_selected H5I_DATASET STRING   16</span></span>
<span><span class="co">## 4     /     mut_table   H5I_GROUP</span></span></code></pre>
</div>
<div class="section level2">
<h2 id="selecting-cells">Selecting cells<a class="anchor" aria-label="anchor" href="#selecting-cells"></a>
</h2>
<p>We are only interested in annotated good-quality cells.</p>
<p>So we will select the cells with annotation, which are good quality
cells.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">f</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"mini_dataset_cell_ann.csv"</span>, package <span class="op">=</span> <span class="st">"scMitoMut"</span><span class="op">)</span></span>
<span><span class="va">cell_ann</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.csv</a></span><span class="op">(</span><span class="va">f</span>, row.names <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span>
<span><span class="co">## Subset the cells, the cell id can be found by colnames() for the Seurat object</span></span>
<span><span class="va">x</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/subset_cell.html">subset_cell</a></span><span class="op">(</span><span class="va">x</span>, <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">cell_ann</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p>After subsetting the cells, the <code>cell_selected</code> slot will
be updated. Only the selected cells will be used in the following
p-value calculation.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">x</span><span class="op">$</span><span class="va">cell_selected</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] "AAACGAAAGAACCCGA-1" "AAACGAAAGTACCTCA-1" "AAACGAACAGAAAGAG-1"</span></span>
<span><span class="co">## [4] "AAACGAAGTGGTCGAA-1" "AAACGAATCAATCGTG-1" "AAACGAATCCCACGGA-1"</span></span></code></pre>
<p>Similarly, we can select loci by using the <code>subset_locus</code>
function. It saves time when we only focus on a few loci.</p>
</div>
<div class="section level2">
<h2 id="calculating-mutation-p-value">Calculating mutation p-value<a class="anchor" aria-label="anchor" href="#calculating-mutation-p-value"></a>
</h2>
<p>Assuming an mtDNA locus’s majority base has constant allele frequency
in cells, the allele frequency mutation was caused by sequencing error
or sampling. Based on that assumption, we build a null hypothesis that
there are no mutations for that locus in the cell. Then we fit the
allele frequency distribution and calculate the probability of observing
allele frequency for a specific locus in a cell. If the probability is
small, we can reject the null hypothesis and conclude that there is a
mutation for that locus in the cell.</p>
<p>To calculate the probability value (p-value), we run
<code>run_calling</code> function, which has 2 arguments: -
<code>mtmutObj</code> is the <code>scMitoMut</code> object -
<code>mc.cores</code> is the number of CPU threads to be used</p>
<p>The process will take some time. The output will be stored in the
<code>pval</code> group of the <code>H5</code> file. The result is
stored in the hard drive, instead of in memory. We don’t need to re-run
the mutation calling when loading the <code>H5</code> file next
time.</p>
<p>The mutation calling is based on beta-binomial distribution. The
mutation p-value is the probability that with the null hypothesis: there
are no mutations for that locus in the cell.</p>
<!--Remove the following lines-->
<p><strong>Detail</strong>: For a specific locus, we calculate the
p-value using the following steps. 1. Defining the wild-type allele as
the allele with the highest average AF among cells. 2. Fitting a 2
components binomial-mixture model to the AF of the wild-type allele for
all cells. Using the model, we define the wild-type cells if it has a
probability &gt;= 0.001 to be the wild type. 3. Using those wild-type
cells, we fit the beta-binomial model. 4. At last, based on the model,
we calculate the p-value of observing the AF of the wild-type allele in
the cell.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Run the model fitting</span></span>
<span><span class="fu"><a href="../reference/run_model_fit.html">run_model_fit</a></span><span class="op">(</span><span class="va">x</span>, mc.cores <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] "chrM.200"</span></span>
<span><span class="co">## [1] "chrM.204"</span></span>
<span><span class="co">## [1] "chrM.310"</span></span>
<span><span class="co">## [1] "chrM.824"</span></span>
<span><span class="co">## [1] "chrM.1000"</span></span>
<span><span class="co">## [1] "chrM.1001"</span></span>
<span><span class="co">## [1] "chrM.1227"</span></span>
<span><span class="co">## [1] "chrM.2285"</span></span>
<span><span class="co">## [1] "chrM.6081"</span></span>
<span><span class="co">## [1] "chrM.9429"</span></span>
<span><span class="co">## [1] "chrM.9728"</span></span>
<span><span class="co">## [1] "chrM.9804"</span></span>
<span><span class="co">## [1] "chrM.9840"</span></span>
<span><span class="co">## [1] "chrM.12889"</span></span>
<span><span class="co">## [1] "chrM.16093"</span></span>
<span><span class="co">## [1] "chrM.16147"</span></span></code></pre>
<pre><code><span><span class="co">##           used (Mb) gc trigger  (Mb) limit (Mb) max used  (Mb)</span></span>
<span><span class="co">## Ncells 1166717 62.4    2296245 122.7         NA  2296245 122.7</span></span>
<span><span class="co">## Vcells 2225694 17.0    8388608  64.0      16384  6295545  48.1</span></span></code></pre>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## The p-value is kept in the pval group of H5 file</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/rhdf5/man/h5ls.html" class="external-link">h5ls</a></span><span class="op">(</span><span class="va">x</span><span class="op">$</span><span class="va">h5f</span>, recursive <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##   group          name       otype   dclass  dim</span></span>
<span><span class="co">## 0     /     cell_list H5I_DATASET   STRING 1359</span></span>
<span><span class="co">## 1     / cell_selected H5I_DATASET   STRING 1359</span></span>
<span><span class="co">## 2     /      loc_list H5I_DATASET   STRING   16</span></span>
<span><span class="co">## 3     /  loc_selected H5I_DATASET   STRING   16</span></span>
<span><span class="co">## 4     /  model_par_bb H5I_DATASET COMPOUND   16</span></span>
<span><span class="co">## 5     /  model_par_bm H5I_DATASET COMPOUND   16</span></span>
<span><span class="co">## 6     /     mut_table   H5I_GROUP              </span></span>
<span><span class="co">## 7     /          pval   H5I_GROUP</span></span></code></pre>
</div>
<div class="section level2">
<h2 id="filter-mutations">Filter mutations<a class="anchor" aria-label="anchor" href="#filter-mutations"></a>
</h2>
<p>Then we will filter the mutations by using the
<code>mut_filter</code> function with the following criteria: - The
mutation has at least 5 cells mutant. - The FDR-adjusted p-value is less
than 0.05.</p>
<p>The output is a <code>data.frame</code> with 2 columns -
<code>loc</code> is the locus - <code>mut_cell_n</code> is the cell
number</p>
<p>We can see that there are 12 loci after filtering.</p>
<p><strong>Detail</strong>: The <code>mut_filter</code> function has 4
arguments: - <code>mtmutObj</code> is the <code>mtmutObj</code> object -
<code>min_cell</code> is the minimum number of mutant cells -
<code>p_adj_method</code> is the method used to adjust the p-value. -
<code>p_threshold</code> is the adjusted p-value threshold</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Filter mutation</span></span>
<span><span class="va">x</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/filter_loc.html">filter_loc</a></span><span class="op">(</span></span>
<span>  mtmutObj <span class="op">=</span> <span class="va">x</span>,</span>
<span>  min_cell <span class="op">=</span> <span class="fl">2</span>,</span>
<span>  model <span class="op">=</span> <span class="st">"bb"</span>,</span>
<span>  p_threshold <span class="op">=</span> <span class="fl">0.01</span>,</span>
<span>  p_adj_method <span class="op">=</span> <span class="st">"fdr"</span></span>
<span><span class="op">)</span></span>
<span><span class="va">x</span><span class="op">$</span><span class="va">loc_pass</span></span></code></pre></div>
<pre><code><span><span class="co">##  [1] "chrM.200"   "chrM.204"   "chrM.310"   "chrM.824"   "chrM.1227" </span></span>
<span><span class="co">##  [6] "chrM.2285"  "chrM.6081"  "chrM.9429"  "chrM.9728"  "chrM.9804" </span></span>
<span><span class="co">## [11] "chrM.9840"  "chrM.12889" "chrM.16093" "chrM.16147"</span></span></code></pre>
</div>
<div class="section level2">
<h2 id="visualization">Visualization<a class="anchor" aria-label="anchor" href="#visualization"></a>
</h2>
<p>We will visualize the mutation by heatmap using the
<code>plot_heatmap</code> function. It can draw a heatmap of p-value,
AF, or binarized mutation status. Its input is the <code>mtmutObj</code>
object. It will independently apply all the filters we used in the
<code>mut_filter</code> function, and select the cells and loci that
pass the filter criteria. In all kinds of figures, the mutation status
will be calculated, and the loci and cells are ordered by the mutation
status.</p>
<p><strong>Detail</strong>: The <code>plot_heatmap</code> arguments. -
<code>mtmutObj</code> is the <code>scMitoMut</code> object -
<code>pos_list</code> is the list of loci - <code>cell_ann</code> is the
cell annotation - <code>ann_colors</code> is the color of the cell
annotation - <code>type</code> is the type of the heatmap which can be
<code>p</code>, <code>af</code>, or <code>binary</code> -
<code>p_adj_method</code> is the method used to adjust the p-value -
<code>p_threshold</code> is the adjusted p-value threshold to determine
if a cell has mutation when selecting the cells and loci -
<code>min_cell_n</code> is the minimum number of cells that have
mutation when selecting the cells and loci - <code>p_binary</code> is
the adjusted p-value threshold to get the binary mutation status -
<code>percent_interp</code> is the percentage overlap threshold between
mutations, to determine if two mutations are correlated for
interpolating the mutation status - <code>n_interp</code> is the number
of overlapped cells to determine if two mutations are correlated for
interpolating</p>
<p>The interpolation is based on the assumption that the mutation are
unique, it is rare to have two mutation in the same population.
Therefore, when two mutations are correlated, one of them is likely a
subclone of the other one.</p>
<div class="section level3">
<h3 id="binary-heatmap">Binary heatmap<a class="anchor" aria-label="anchor" href="#binary-heatmap"></a>
</h3>
<p>The binary heatmap will show the mutation status of each cell for
each locus. The red means mutant, the blue means no mutant, and the grey
means the missing value.</p>
<p>For the mutation status, if two mutations are correlated, we set the
high-frequency locus will get the mutant status from the low-frequency
one. The correlation is determined by the percentage and number of cells
overlapping between the two mutations.</p>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Prepare the color for cell annotation</span></span>
<span><span class="va">colors</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span></span>
<span>  <span class="st">"Cancer Epi"</span> <span class="op">=</span> <span class="st">"#f28482"</span>,</span>
<span>  Blood <span class="op">=</span> <span class="st">"#f6bd60"</span></span>
<span><span class="op">)</span></span>
<span><span class="va">ann_colors</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="st">"SeuratCellTypes"</span> <span class="op">=</span> <span class="va">colors</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## binary heatmap</span></span>
<span><span class="fu"><a href="../reference/plot_heatmap.html">plot_heatmap</a></span><span class="op">(</span><span class="va">x</span>,</span>
<span>  cell_ann <span class="op">=</span> <span class="va">cell_ann</span>, ann_colors <span class="op">=</span> <span class="va">ann_colors</span>, type <span class="op">=</span> <span class="st">"binary"</span>,</span>
<span>  percent_interp <span class="op">=</span> <span class="fl">0.2</span>, n_interp <span class="op">=</span> <span class="fl">3</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="Analysis_colon_cancer_dataset_files/figure-html/unnamed-chunk-12-1.png" width="1152"></p>
<p>Also we can turn off the interpolation by setting
<code>percent_interp = 1</code>.</p>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## binary heatmap</span></span>
<span><span class="fu"><a href="../reference/plot_heatmap.html">plot_heatmap</a></span><span class="op">(</span><span class="va">x</span>,</span>
<span>  cell_ann <span class="op">=</span> <span class="va">cell_ann</span>, ann_colors <span class="op">=</span> <span class="va">ann_colors</span>, type <span class="op">=</span> <span class="st">"binary"</span>,</span>
<span>  percent_interp <span class="op">=</span> <span class="fl">1</span>, n_interp <span class="op">=</span> <span class="fl">3</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="Analysis_colon_cancer_dataset_files/figure-html/unnamed-chunk-13-1.png" width="1152"></p>
</div>
<div class="section level3">
<h3 id="p-value-heatmap">P value heatmap<a class="anchor" aria-label="anchor" href="#p-value-heatmap"></a>
</h3>
<p>The p-value heatmap shows the adjusted p-value of each cell for each
locus. The order of the cells and loci is determined by the mutation
status.</p>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## p value heatmap</span></span>
<span><span class="fu"><a href="../reference/plot_heatmap.html">plot_heatmap</a></span><span class="op">(</span><span class="va">x</span>,</span>
<span>  cell_ann <span class="op">=</span> <span class="va">cell_ann</span>, ann_colors <span class="op">=</span> <span class="va">ann_colors</span>, type <span class="op">=</span> <span class="st">"p"</span>,</span>
<span>  percent_interp <span class="op">=</span> <span class="fl">0.2</span>, n_interp <span class="op">=</span> <span class="fl">3</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="Analysis_colon_cancer_dataset_files/figure-html/unnamed-chunk-14-1.png" width="1152"></p>
</div>
<div class="section level3">
<h3 id="af-heatmap">AF heatmap<a class="anchor" aria-label="anchor" href="#af-heatmap"></a>
</h3>
<p>The AF heatmap will show the AF of each cell for each locus. Similar
to the p-value heatmap, the order of the cells and loci are determined
by the mutation status.</p>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## allele frequency heatmap</span></span>
<span><span class="fu"><a href="../reference/plot_heatmap.html">plot_heatmap</a></span><span class="op">(</span><span class="va">x</span>,</span>
<span>  cell_ann <span class="op">=</span> <span class="va">cell_ann</span>, ann_colors <span class="op">=</span> <span class="va">ann_colors</span>, type <span class="op">=</span> <span class="st">"af"</span>,</span>
<span>  percent_interp <span class="op">=</span> <span class="fl">0.2</span>, n_interp <span class="op">=</span> <span class="fl">3</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="Analysis_colon_cancer_dataset_files/figure-html/unnamed-chunk-15-1.png" width="1152"></p>
</div>
</div>
<div class="section level2">
<h2 id="exporting-mutation-data">Exporting mutation data<a class="anchor" aria-label="anchor" href="#exporting-mutation-data"></a>
</h2>
<p>We can export the mutation data by using the following functions:</p>
<ul>
<li>
<code>export_df</code> export the mutation data as a
<code>data.frame</code>
</li>
<li>
<code>export_af</code> export the AF data as a
<code>data.matrix</code> with loci as row names and cells as column
names.</li>
<li>
<code>export_pval</code> export the p-value data as a
<code>data.matrix</code> with loci as row names and cells as column
names.</li>
<li>
<code>export_binary</code> export the mutation status data as a
<code>data.matrix</code> with loci as row names and cells as column
names.</li>
</ul>
<p>Those functions have the same filtering options as the
<code>plot_heatmap</code> function.</p>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Export the mutation data as data.frame</span></span>
<span><span class="va">m_df</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/export_dt.html">export_df</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span></span>
<span><span class="va">m_df</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">10</span>, <span class="op">]</span></span></code></pre></div>
<pre><code><span><span class="co">##          loc       cell_barcode alt_depth fwd_depth rev_depth coverage pval</span></span>
<span><span class="co">## 1  chrM.1227 AAACGAAAGAACCCGA-1        90        48        42       91    1</span></span>
<span><span class="co">## 2  chrM.1227 AAACGAAAGTACCTCA-1        49        23        26       49    1</span></span>
<span><span class="co">## 3  chrM.1227 AAACGAACAGAAAGAG-1        60        25        35       60    1</span></span>
<span><span class="co">## 4  chrM.1227 AAACGAAGTGGTCGAA-1        52        25        27       52    1</span></span>
<span><span class="co">## 5  chrM.1227 AAACTCGAGGTCGGTA-1        25        15        10       25    1</span></span>
<span><span class="co">## 6  chrM.1227 AAACTCGCAGTGGTCC-1        36        21        15       37    1</span></span>
<span><span class="co">## 7  chrM.1227 AAACTCGTCGGGCTCA-1        49        24        25       49    1</span></span>
<span><span class="co">## 8  chrM.1227 AAACTGCGTGAGGGTT-1        54        27        27       54    1</span></span>
<span><span class="co">## 9  chrM.1227 AAAGATGAGTCGCCTG-1        77        32        45       77    1</span></span>
<span><span class="co">## 10 chrM.1227 AAAGATGCAGGAGCAT-1        27        15        12       27    1</span></span>
<span><span class="co">##          af mut_status</span></span>
<span><span class="co">## 1  0.989011      FALSE</span></span>
<span><span class="co">## 2  1.000000      FALSE</span></span>
<span><span class="co">## 3  1.000000      FALSE</span></span>
<span><span class="co">## 4  1.000000      FALSE</span></span>
<span><span class="co">## 5  1.000000      FALSE</span></span>
<span><span class="co">## 6  0.972973      FALSE</span></span>
<span><span class="co">## 7  1.000000      FALSE</span></span>
<span><span class="co">## 8  1.000000      FALSE</span></span>
<span><span class="co">## 9  1.000000      FALSE</span></span>
<span><span class="co">## 10 1.000000      FALSE</span></span></code></pre>
<div class="sourceCode" id="cb27"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Export allele frequency data as data.matrix</span></span>
<span><span class="fu"><a href="../reference/export_dt.html">export_af</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">5</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">5</span><span class="op">]</span></span></code></pre></div>
<pre><code><span><span class="co">##            AAACGAAAGAACCCGA-1 AAACGAAAGTACCTCA-1 AAACGAACAGAAAGAG-1</span></span>
<span><span class="co">## chrM.1227           0.9890110                  1                  1</span></span>
<span><span class="co">## chrM.12889          0.9821429                  1                  1</span></span>
<span><span class="co">## chrM.16093          1.0000000                  1                  1</span></span>
<span><span class="co">## chrM.16147          0.3606557                  0                  0</span></span>
<span><span class="co">## chrM.200            1.0000000                  1                  1</span></span>
<span><span class="co">##            AAACGAAGTGGTCGAA-1 AAACTCGAGGTCGGTA-1</span></span>
<span><span class="co">## chrM.1227                   1                  1</span></span>
<span><span class="co">## chrM.12889                  1                  1</span></span>
<span><span class="co">## chrM.16093                  1                  1</span></span>
<span><span class="co">## chrM.16147                  0                  1</span></span>
<span><span class="co">## chrM.200                    1                  1</span></span></code></pre>
<div class="sourceCode" id="cb29"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Export p-value data as data.matrix</span></span>
<span><span class="fu"><a href="../reference/export_dt.html">export_pval</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">5</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">5</span><span class="op">]</span></span></code></pre></div>
<pre><code><span><span class="co">##            AAACGAAAGAACCCGA-1 AAACGAAAGTACCTCA-1 AAACGAACAGAAAGAG-1</span></span>
<span><span class="co">## chrM.1227        1.000000e+00        1.00000e+00       1.000000e+00</span></span>
<span><span class="co">## chrM.12889       3.559443e-01        1.00000e+00       1.000000e+00</span></span>
<span><span class="co">## chrM.16093       1.000000e+00        1.00000e+00       1.000000e+00</span></span>
<span><span class="co">## chrM.16147       2.972594e-10        1.95334e-12       8.299404e-13</span></span>
<span><span class="co">## chrM.200         1.000000e+00        1.00000e+00       1.000000e+00</span></span>
<span><span class="co">##            AAACGAAGTGGTCGAA-1 AAACTCGAGGTCGGTA-1</span></span>
<span><span class="co">## chrM.1227        1.000000e+00                  1</span></span>
<span><span class="co">## chrM.12889       1.000000e+00                  1</span></span>
<span><span class="co">## chrM.16093       1.000000e+00                  1</span></span>
<span><span class="co">## chrM.16147       1.082287e-09                  1</span></span>
<span><span class="co">## chrM.200         1.000000e+00                  1</span></span></code></pre>
<div class="sourceCode" id="cb31"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Export binary mutation status data as data.matrix</span></span>
<span><span class="fu"><a href="../reference/export_dt.html">export_binary</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">5</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">5</span><span class="op">]</span></span></code></pre></div>
<pre><code><span><span class="co">##            AAACGAAAGAACCCGA-1 AAACGAAAGTACCTCA-1 AAACGAACAGAAAGAG-1</span></span>
<span><span class="co">## chrM.1227                   0                  0                  0</span></span>
<span><span class="co">## chrM.12889                  0                  0                  0</span></span>
<span><span class="co">## chrM.16093                  0                  0                  0</span></span>
<span><span class="co">## chrM.16147                  1                  1                  1</span></span>
<span><span class="co">## chrM.200                    0                  0                  0</span></span>
<span><span class="co">##            AAACGAAGTGGTCGAA-1 AAACTCGAGGTCGGTA-1</span></span>
<span><span class="co">## chrM.1227                   0                  0</span></span>
<span><span class="co">## chrM.12889                  0                  0</span></span>
<span><span class="co">## chrM.16093                  0                  0</span></span>
<span><span class="co">## chrM.16147                  1                  0</span></span>
<span><span class="co">## chrM.200                    0                  0</span></span></code></pre>
</div>
<div class="section level2">
<h2 id="show-the-p-value-versus-af-plot">Show the p value versus af plot<a class="anchor" aria-label="anchor" href="#show-the-p-value-versus-af-plot"></a>
</h2>
<p>Lastly, we try to show the distribution of p value and AF value. It
is also a demonstration of how to extract the information from the H5
file and use it for further analysis.</p>
<div class="sourceCode" id="cb33"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">m_dt</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://Rdatatable.gitlab.io/data.table/reference/data.table.html" class="external-link">data.table</a></span><span class="op">(</span><span class="va">m_df</span><span class="op">)</span></span>
<span><span class="va">m_dt</span><span class="op">[</span>, <span class="va">cell_type</span> <span class="op">:=</span> <span class="va">cell_ann</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/character.html" class="external-link">as.character</a></span><span class="op">(</span><span class="va">m_dt</span><span class="op">$</span><span class="va">cell_barcode</span><span class="op">)</span>, <span class="st">"SeuratCellTypes"</span><span class="op">]</span><span class="op">]</span></span>
<span><span class="va">m_dt_sub</span> <span class="op">&lt;-</span> <span class="va">m_dt</span><span class="op">[</span><span class="va">loc</span> <span class="op">==</span> <span class="st">"chrM.1227"</span><span class="op">]</span></span>
<span><span class="va">m_dt_sub</span><span class="op">[</span>, <span class="fu"><a href="https://rdrr.io/r/base/sum.html" class="external-link">sum</a></span><span class="op">(</span><span class="op">(</span><span class="va">pval</span><span class="op">)</span> <span class="op">&lt;</span> <span class="fl">0.01</span>, na.rm <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>, by <span class="op">=</span> <span class="va">cell_type</span><span class="op">]</span></span></code></pre></div>
<pre><code><span><span class="co">##     cell_type V1</span></span>
<span><span class="co">## 1: Cancer Epi 28</span></span>
<span><span class="co">## 2:      Blood  0</span></span></code></pre>
<div class="sourceCode" id="cb35"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">m_dt_sub</span><span class="op">[</span>, <span class="fu"><a href="https://rdrr.io/r/base/sum.html" class="external-link">sum</a></span><span class="op">(</span><span class="op">(</span><span class="fl">1</span> <span class="op">-</span> <span class="va">af</span><span class="op">)</span> <span class="op">&gt;</span> <span class="fl">0.05</span>, na.rm <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>, by <span class="op">=</span> <span class="va">cell_type</span><span class="op">]</span></span></code></pre></div>
<pre><code><span><span class="co">##     cell_type V1</span></span>
<span><span class="co">## 1: Cancer Epi 42</span></span>
<span><span class="co">## 2:      Blood  2</span></span></code></pre>
<div class="sourceCode" id="cb37"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="va">m_dt_sub</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">cell_type</span>, y <span class="op">=</span> <span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log10</a></span><span class="op">(</span><span class="va">pval</span><span class="op">)</span>, color <span class="op">=</span> <span class="va">cell_type</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_jitter.html" class="external-link">geom_jitter</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_manual.html" class="external-link">scale_color_manual</a></span><span class="op">(</span>values <span class="op">=</span> <span class="va">colors</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html" class="external-link">theme_bw</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_abline.html" class="external-link">geom_hline</a></span><span class="op">(</span>yintercept <span class="op">=</span> <span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log10</a></span><span class="op">(</span><span class="fl">0.01</span><span class="op">)</span>, linetype <span class="op">=</span> <span class="st">"dashed"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ylab</a></span><span class="op">(</span><span class="st">"-log10(FDR)"</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Warning in FUN(X[[i]], ...): NaNs produced</span></span>
<span></span>
<span><span class="co">## Warning in FUN(X[[i]], ...): NaNs produced</span></span></code></pre>
<pre><code><span><span class="co">## Warning: Removed 4 rows containing missing values (`geom_point()`).</span></span></code></pre>
<p><img src="Analysis_colon_cancer_dataset_files/figure-html/unnamed-chunk-17-1.png" width="480"></p>
<div class="sourceCode" id="cb40"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="va">m_dt_sub</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">cell_type</span>, y <span class="op">=</span> <span class="fl">1</span> <span class="op">-</span> <span class="va">af</span>, color <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">factor</a></span><span class="op">(</span><span class="va">cell_type</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_jitter.html" class="external-link">geom_jitter</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_manual.html" class="external-link">scale_color_manual</a></span><span class="op">(</span>values <span class="op">=</span> <span class="va">colors</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html" class="external-link">theme_bw</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_abline.html" class="external-link">geom_hline</a></span><span class="op">(</span>yintercept <span class="op">=</span> <span class="fl">0.05</span>, linetype <span class="op">=</span> <span class="st">"dashed"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ylab</a></span><span class="op">(</span><span class="st">"1 - Dominant Allele Frequency"</span><span class="op">)</span></span></code></pre></div>
<p><img src="Analysis_colon_cancer_dataset_files/figure-html/unnamed-chunk-17-2.png" width="480"></p>
</div>
<div class="section level2">
<h2 id="session-info">Session Info<a class="anchor" aria-label="anchor" href="#session-info"></a>
</h2>
<div class="sourceCode" id="cb41"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## R version 4.3.0 (2023-04-21)</span></span>
<span><span class="co">## Platform: x86_64-apple-darwin20 (64-bit)</span></span>
<span><span class="co">## Running under: macOS 14.1</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Matrix products: default</span></span>
<span><span class="co">## BLAS:   /Library/Frameworks/R.framework/Versions/4.3-x86_64/Resources/lib/libRblas.0.dylib </span></span>
<span><span class="co">## LAPACK: /Library/Frameworks/R.framework/Versions/4.3-x86_64/Resources/lib/libRlapack.dylib;  LAPACK version 3.11.0</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## locale:</span></span>
<span><span class="co">## [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## time zone: Europe/Paris</span></span>
<span><span class="co">## tzcode source: internal</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## attached base packages:</span></span>
<span><span class="co">## [1] stats     graphics  grDevices utils     datasets  methods   base     </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## other attached packages:</span></span>
<span><span class="co">## [1] rhdf5_2.44.0      ggplot2_3.4.2     data.table_1.14.8 scMitoMut_0.99.1 </span></span>
<span><span class="co">## [5] knitr_1.43        BiocStyle_2.28.0 </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## loaded via a namespace (and not attached):</span></span>
<span><span class="co">##  [1] sass_0.4.6          utf8_1.2.3          generics_0.1.3     </span></span>
<span><span class="co">##  [4] stringi_1.7.12      hms_1.1.3           digest_0.6.31      </span></span>
<span><span class="co">##  [7] magrittr_2.0.3      RColorBrewer_1.1-3  evaluate_0.21      </span></span>
<span><span class="co">## [10] grid_4.3.0          bookdown_0.34       fastmap_1.1.1      </span></span>
<span><span class="co">## [13] R.oo_1.25.0         plyr_1.8.8          rprojroot_2.0.3    </span></span>
<span><span class="co">## [16] jsonlite_1.8.5      R.utils_2.12.2      BiocManager_1.30.21</span></span>
<span><span class="co">## [19] purrr_1.0.1         fansi_1.0.4         scales_1.2.1       </span></span>
<span><span class="co">## [22] textshaping_0.3.6   jquerylib_0.1.4     cli_3.6.1          </span></span>
<span><span class="co">## [25] rlang_1.1.1         R.methodsS3_1.8.2   munsell_0.5.0      </span></span>
<span><span class="co">## [28] withr_2.5.0         cachem_1.0.8        yaml_2.3.7         </span></span>
<span><span class="co">## [31] parallel_4.3.0      tools_4.3.0         tzdb_0.4.0         </span></span>
<span><span class="co">## [34] memoise_2.0.1       dplyr_1.1.2         colorspace_2.1-0   </span></span>
<span><span class="co">## [37] Rhdf5lib_1.22.0     vctrs_0.6.3         R6_2.5.1           </span></span>
<span><span class="co">## [40] lifecycle_1.0.3     zlibbioc_1.46.0     stringr_1.5.0      </span></span>
<span><span class="co">## [43] fs_1.6.2            ragg_1.2.5          pkgconfig_2.0.3    </span></span>
<span><span class="co">## [46] desc_1.4.2          pkgdown_2.0.7       bslib_0.5.0        </span></span>
<span><span class="co">## [49] pillar_1.9.0        gtable_0.3.3        glue_1.6.2         </span></span>
<span><span class="co">## [52] Rcpp_1.0.10         systemfonts_1.0.4   highr_0.10         </span></span>
<span><span class="co">## [55] xfun_0.39           tibble_3.2.1        tidyselect_1.2.0   </span></span>
<span><span class="co">## [58] rhdf5filters_1.12.1 farver_2.1.1        htmltools_0.5.5    </span></span>
<span><span class="co">## [61] labeling_0.4.2      rmarkdown_2.22      readr_2.1.4        </span></span>
<span><span class="co">## [64] pheatmap_1.0.12     compiler_4.3.0</span></span></code></pre>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Wenjie Sun, Leila Perie.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
